keps/sig-node/4205-psi-metric/README.md

```diff
@@ -0,0 +1,637 @@
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Phase 1: Alpha
+
+- PSI integrated in kubelet behind a feature flag.
+- Initial e2e tests completed and enabled.
+
+#### Phase 2: Alpha
+
+- Implement Phase 2 of the enhancement which enables kubelet to 
+report node conditions based off PSI values.
+- Additional tests coverage.
+
+#### Beta
+
+- Feature gate is enabled by default.
```

Comment:
It complicates things, but it would be nice to have two feature gates.  One for the reporting of the PSI in status that could be on by default earlier than a second feature gate for turning that reporting into node conditions. This would allow some more visibility into PSI before committing to default thresholds. Not sure if there is precedent for this though.

Reply:
We could definitely add two feature gates, but IMO adding PSI metrics to stats wouldn't impact anything as such. I am trying to understand the usecase where user would want to disable PSI metrics in stats API. It will definitely help with visibility, but it still wouldn't provide any benefits or visibility for phase 2 of this KEP to set node conditions based on PSI. Thoughts?

Reply:
Ah, I didn't realize you were planning on having them on by default without a feature gate.  That works for me as well and eliminates the need for a separate feature gate.